src/terminal-search-dialog.c

```diff
@@ -350,13 +353,13 @@ terminal_search_dialog_get_regex (GtkWidget *dialog)
 
 	pattern = text = terminal_search_dialog_get_search_text (dialog);
 
-	compile_flags = G_REGEX_OPTIMIZE;
+	compile_flags = PCRE2_UTF | PCRE2_NO_UTF_CHECK;
 
 	if (!GET_FLAG (match_case_checkbutton))
-		compile_flags |= G_REGEX_CASELESS;
+		compile_flags |= PCRE2_CASELESS;
 
 	if (GET_FLAG (regex_checkbutton))
-		compile_flags |= G_REGEX_MULTILINE;
+		compile_flags |= PCRE2_MULTILINE;
 	else
 		pattern = g_regex_escape_string (text, -1);
```

Comment:
In addition (`GET_FLAG (regex_checkbutton) == false`)
```
compile_flags |= PCRE2_LITERAL
```

Reply:
Should that be instead of the `g_regex_escape_string` I guess?

Reply:
It's not necessary, because the user can select "Match entire word only".
`pattern := "\bESCAPED-TEXT\b"`

https://manpages.debian.org/stretch/libpcre2-dev/pcre2pattern.3.en.html

https://github.com/mate-desktop/mate-terminal/blob/99f9ea27efe6376e3137f2942032f76937d6e603/src/terminal-search-dialog.c#L369

Reply:
So I checked, adding `PCRE2_LITERAL` here breaks search functionality, independent of the escaping (with or without, both tested, search are broken).

OTOH not adding the LITERAL makes search work and the error pops up.

So are you sure we need to use `LITERAL` here?

Reply:
I confirm that if the PCRE2_LITERAL flag is set the search won't work anymore.